### PR TITLE
Typescript definitions

### PR DESCRIPTION
**Pull request by dan-silver**
Generates https://github.com/microsoftgraph/msgraph-typescript-typings/blob/master/microsoft-graph.d.ts

### CODE REVIEW

File: src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            foreach (var member in _enum.Members)
+            {
+                if (enumVals == null)
+                {
+                    enumVals = '"' + member.Name + '"';
+                } else
+                {
+                    enumVals += " | " + '"' + member.Name + '"';
+                }
+            }
```
Comment (shiftylogic) [inline src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs]: This entire function can be replaced with:
  return _enum.Members.Select(s => "\"" + s + "\"").Join(" | ");

File: src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    break;
+                case "Boolean":
+                    typeStr = "boolean";
+                    break;
+
+            }
+            return (prop.IsCollection) ? "[" + typeStr + "]" : typeStr;
+            
+        }
+        public static String UpperCaseFirstChar(String s)
```
Comment (shiftylogic) [inline src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs]: Why is this exact function in 2 different classes?

File: src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+            }
+
+            @namespace = ConfigurationService.Settings.NamespaceOverride;
+            return folderName != null ? string.Format("{0}.{1}", @namespace, folderName)
+                                         : @namespace;
+        }
+
+        private string CreatePathFromNamespace(string @namespace)
+        {
+            var splittedPaths = @namespace.Split('.');
```
Comment (shiftylogic) [inline src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs]: "splitPaths"?

Comment (dan-silver) [inline src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs]: This code was copy-pasted from another path writer. It looks like they all have this duplicate code unfortunately. I'd prefer for the TS file to not diverge from the others so when it's cleaned up we can easily find duplicates.

File: src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs (hunk 1 above)
Comment (shiftylogic) [inline src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs]: Sorry. Aggregate not join, and the syntax is slightly different. I was crossing my .NET with some JavaScript.

File: src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs (hunk 2 above)
Comment (dan-silver) [inline src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs]: One was referenced in t4 through the code writer and the other was a private helper in the TypeHelper. I've refactored & merged them.

File: src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs (hunk 3 above)
Comment (dan-silver) [inline src/GraphODataTemplateWriter/PathWriters/TypeScriptPathWriter.cs]: Just following the pattern of the other templates :) I've updated the name

File: src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs (hunk 1 above)
Comment (dan-silver) [inline src/GraphODataTemplateWriter/CodeHelpers/TypeScript/TypeHelperTypeScript.cs]: Thanks! Got it working with return _enum.Members.Select(m => "\"" + m.Name + "\"").Aggregate((cur, next) =>  cur + " | " + next);

**Pull request merged**